cetmix_tower_server/models/cx_tower_plan.py

```diff
@@ -99,7 +99,9 @@ def _execute_single(self, server, **kwargs):
         plan_log_obj = self.env["cx.tower.plan.log"].sudo()
 
         # Check if the same plan is being executed on this server right now
-        if not self.allow_parallel_run:
+        if not self.allow_parallel_run or self.env.context.get(
+            "disable_recursive_plan"
```

Comment:
Better to rename it into something more relevant. Eg `prevent_plan_recursion`.
Otherwise it may look like we can disable or enable recursion which is not true.